### ISSUE TITLE
delete element.dataset.keyname throws TypeError: 'deleteProperty' on proxy

### ISSUE THREAD

**Issue by WebMechanic** [capricorn86/happy-dom#652]
**Describe the bug**

To remove an attribute from an element's dataset, you should be able to [use the delete operator](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset#setting_values): `delete element.dataset.keyname` whether `.keyname` exists or not.

If `.keyname` does not already exist on the dataset, the statement throws with a
`TypeError: 'deleteProperty' on proxy: trap returned falsish for property 'keyname'`.

Native DOM allows to do so and even returns `true`.

**To Reproduce**
inside a Browser with native DOM
```js
const elt = document.body;
delete elt.dataset.fubar;  // true
```

inside a unit test with Happy DOM
```js
const elt = document.body;
delete elt.dataset.fubar;     // throws
delete elt.dataset['fubar'];  // throws too
```
same but keys are initialised first with "anything"
```js
const elt = document.body;
elt.dataset.foo = null;
delete elt.dataset.foo;  // works

elt.dataset.bar = undefined;
delete elt.dataset.bar;  // works too
```

**Expected behavior**
To not throw a TypeError when attempting to `delete` a non-existing dataset key, but silently ignore that the key does not exist.

**Additional context**
 - Node 16.18.0
 - Happy DOM 7.6.6
 - Vite 3.2.3 
 - Vitest vitest/0.24.5

I'm writing a utility to manage multiple values inside data attributes (data-fubar="foo bar bat baz") similar to `classList` . Between tests I wanted to clean `dataset` and ran into this issue.
Not a showstopper 'cos I can just used a different key for each suite.

**Comment by takaya1992** [comment]
`delete` operator ( deleteProperty ) is always returns `true`, except under exceptional conditions.

> `true` for all cases except when the property is an own non-configurable property, in which case false is returned in non-strict mode.
> 
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete#return_value

but, this code returns false when the property not exist, so there is a difference in behavior.

https://github.com/capricorn86/happy-dom/blob/67d95f012f9c89e717b30a24e7e3f5bbf28266ac/packages/happy-dom/src/nodes/html-element/HTMLElement.ts#L252-L258

**Comment by btea** [comment]
@capricorn86 This issue has been fixed and should be able to be closed.

**Comment by sasha-evidation** [comment]
@btea This appears to not be fixed as of the latest version (10.11.2). Here's a very simple case that I can reproduce in a Jest test (using the Happy DOM Jest environment):

```javascript
test("is able to delete a non-existent dataset key", () => {
  delete document.documentElement.dataset.xyz
})
```

Fails with:

```
TypeError: 'deleteProperty' on proxy: trap returned falsish for property 'xyz'
```

**Comment by capricorn86** [comment]
Thank you for reporting @WebMechanic, @takaya1992, @btea and @sasha-evidation! :slightly_smiling_face: 

Big thanks to @RussianCow for the contribution :star: 

There is a fix out now:
https://github.com/capricorn86/happy-dom/releases/tag/v12.1.7
